src/libs/CurrencyUtils.ts

```diff
@@ -96,11 +96,11 @@ function convertToFrontendAmountAsInteger(amountAsInt: number): number {
  *
  * @note we do not support any currencies with more than two decimal places.
  */
-function convertToFrontendAmountAsString(amountAsInt: number | null | undefined): string {
+function convertToFrontendAmountAsString(amountAsInt: number | null | undefined, currency?: string): string {
     if (amountAsInt === null || amountAsInt === undefined) {
         return '';
     }
-    return convertToFrontendAmountAsInteger(amountAsInt).toFixed(2);
+    return convertToFrontendAmountAsInteger(amountAsInt).toFixed(currency ? getCurrencyDecimals(currency) : 2);
```

Comment:
```suggestion
    return convertToFrontendAmountAsInteger(amountAsInt).toFixed(getCurrencyDecimals(currency));
```

If there is no currency, it will default to USD, which has two decimal places

Reply:
correct.